x-pack/plugins/fleet/public/components/agent_enrollment_flyout/confirm_incoming_data.tsx

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useEffect, useState } from 'react';
+import { EuiCallOut, EuiText, EuiSpacer } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+
+import type { IncomingDataList } from '../../../public/applications/fleet/types';
+import { sendGetAgentIncomingData } from '../../hooks';
+interface Props {
+  agentsIds: string[];
+}
+
+export const ConfirmIncomingData: React.FunctionComponent<Props> = ({ agentsIds }) => {
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+  const [incomingData, setIncomingData] = useState<IncomingDataList[]>([]);
+
+  useEffect(() => {
+    const getIncomingData = async () => {
+      const { data } = await sendGetAgentIncomingData({ agentsIds });
+      if (data?.items) {
+        setIncomingData(data?.items);
+        setIsLoading(false);
+      }
+    };
+    if (agentsIds) {
+      getIncomingData();
+    }
+  }, [agentsIds]);
+
+  const enrolledAgents = incomingData.length;
+  const agentsWithData = incomingData.reduce((acc, curr) => {
+    const agentData = Object.values(curr)[0];
+    return !!agentData.data ? acc + 1 : acc;
+  }, 0);
```

Comment:
nit: I think it wil make sense to move this to his own hook function make things a little more separate and it could allow to write test for the hook or the display component easily in the future

Reply:
Done, I split it up the component and moved all the logic in a separate hook.